integration/benchnet2/automate/bootstrap_automation.go

```diff
@@ -0,0 +1,196 @@
+	var collectionNodes = NodesDefs{Defaults: *nodeTypeResources, Nodes: collectionNodeMap}
+
+	consensusTemplatePath := templateFolder + CONSENSUS_TEMPLATE
+	consensusNodeMap := nodeStruct("consensus", consensusTemplatePath, DEFAULT_CONSENSUS_IMAGE, nodesData, nodeConfig)
+	var consensusNodes = NodesDefs{Defaults: *nodeTypeResources, Nodes: consensusNodeMap}
+
+	executionTemplatePath := templateFolder + EXECUTION_TEMPLATE
+	executionNodeMap := nodeStruct("execution", executionTemplatePath, DEFAULT_EXECUTION_IMAGE, nodesData, nodeConfig)
+	var executionNodes = NodesDefs{Defaults: *nodeTypeResources, Nodes: executionNodeMap}
+
+	verificationTemplatePath := templateFolder + VERIFICATION_TEMPLATE
+	verificationNodeMap := nodeStruct("verification", verificationTemplatePath, DEFAULT_VERIFICATION_IMAGE, nodesData, nodeConfig)
+	var verificationNodes = NodesDefs{Defaults: *nodeTypeResources, Nodes: verificationNodeMap}
+
+	return &Values{Branch: "fake-branch", Commit: "123456", Defaults: EmptyStruct{}, Access: accessNodes, Collection: collectionNodes,
+		Consensus: consensusNodes, Execution: executionNodes, Verification: verificationNodes}
+}
+
+func nodeStruct(nodeType string, nodeTemplatePath string, imageTag string, nodesData map[string]Node, nodeConfig map[string]int) map[string]NodeDetails {
+	nodeTemplate := createTemplate(nodeTemplatePath)
+	var nodeMap = make(map[string]NodeDetails)
+
+	for i := 1; i <= nodeConfig[nodeType]; i++ {
+		name := fmt.Sprint(nodeType, i)
+		replacementData := ReplacementData{NodeID: nodesData[name].NodeID, ImageTag: imageTag}
+
+		nodeString := replaceTemplateData(nodeTemplate, replacementData)
+		nodeStruct := unmarshalToStruct(nodeString, &NodeDetails{}).(*NodeDetails)
+		nodeStruct.Image = replacementData.ImageTag
+		nodeStruct.NodeID = replacementData.NodeID
+
+		nodeMap[name] = *nodeStruct
+	}
+
+	return nodeMap
+}
+
+func textReader(path string) []byte {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return file
+}
+
+func writeYamlBytesToFile(filepath string, yamlData []byte) {
+	file, err := os.OpenFile(filepath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	_, err = file.Write(yamlData)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	file.Close()
+}
+
+func replaceTemplateData(template *template.Template, data ReplacementData) string {
+	var doc bytes.Buffer
+	err := template.Execute(&doc, data)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return doc.String()
+}
+
+func createTemplate(templatePath string) *template.Template {
+	template, err := template.New("todos").Parse(string(textReader(templatePath)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return template
+}
+
+func unmarshalToStruct(source string, target interface{}) interface{} {
+	err := yaml.Unmarshal([]byte(source), target)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return target
+}
+
+func marshalToYaml(source interface{}, outputFilePath string) {
```

Comment:
Could we log the `outpFilePath` that this file will be written to so that the user understands where the file is being written to? This would be valuable for troubleshooting and local testing.